MatiasDure/Project_IMTS
# Passive Event Manager and Hide And Sea Passive Event

**Description**
* Set up a passive event manager
* Created a few helper classes like Cooldown and Frequency to control specific parts of the passive event
* Created the Hide and Sea passive event
* Passive event Config Scriptable Object to control values of passive events

**Visual**
![hideAndSee](https://github.com/user-attachments/assets/c28dfaf8-89ee-4d71-b019-4ef456d1da2b)

**How To Test**
1. Go to the HideAndSeaPassiveEvent scene
2. Rotate to scan the image and spawn the companion
3. Companion should hide in one of the hidings spots provided
4. Play around with the config file which can be found in the _OceanPassiveEvents_ game object in the hierarchy. The file appears as _Config_ in the Hide and Sea script.
5. Make sure all values work as expected
